python.rst

```diff
@@ -761,27 +762,30 @@ Inversion of Control and Dependency Injection
   flexibility (injected objects and resources need only meet an
   interface).
 
-- It’s a trivial one-liner to add object creation to a function that
-  accepts an object as an argument, the converse requires rewriting
-  the function.
+- You can keep your method signatures simple by injecting the
+  dependencies in the constructor. This is a good idea for objects that
+  are instantiated once and used many times. For example, a class
+  might take a gRPC client in its constructor and use it to implement
+  its methods - this lets setup code worry about creating the client,
+  and business logic code can call the methods of the instantiated
+  class without knowing about the client.
 
-Accessors
----------
-
-- Accessors allow property access to be caught and modified later
-  without changing access sites. However, if a significant amount of
-  work (in particular, resource allocation or io) is added around
-  property access, move that functionality to a method.
+- It’s a one-liner to add object creation to a function that
+  accepts an object as an argument; the converse requires rewriting
+  the function.
 
 Constructors
 ------------
 
+- Use `dataclasses <https://docs.python.org/3/library/dataclasses.html>`_
+  to create simple data objects.
+
 - Limit the amount of “real work” done in a constructor. Dependency
   injection is a tremendous help here.
 
 - If an object requires expensive initialization (e.g. the creation of
   a zookeeper session, communication over the network, file IO,
-  concurrency) use a separate method to initialize the object. Also
+  concurrency) use a separate classmethod to initialize the object. Also
```

Comment:
We can highlight composition as a solution to this problem as well. `classmethod` may still create issues with coupling as the function that is used to create the expensive resource will probably become a hard dependency and the `classmethod` may become untestable as well. 
Would suggesting wiring long-lived resources as close to `main()` as possible be overkill? This way a class could be fully unit-testable and the only subject of integration testing are when the classes are wired together.

Reply:
yeah—feel free to use that `add a suggestion` button and rewrite it with what you think it needs, otherwise I can take a swing at it.

me leaving that sort of stuff out is not a reflection of me disagreeing, somuchas me for years having found it impossible to transmit an appreciation of dependency injection via any mechanism other than CR